web/src/app/administration/components/workbasket-access-items/workbasket-access-items.component.spec.ts

```diff
@@ -0,0 +1,194 @@
+        { provide: FormsValidatorService, useClass: formValidatorServiceSpy },
+        { provide: NotificationService, useClass: notificationServiceSpy },
+        ClassificationCategoriesService,
+        WorkbasketService,
+        DomainService,
+        SelectedRouteService,
+        StartupService,
+        TaskanaEngineService,
+        WindowRefService
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(WorkbasketAccessItemsComponent);
+    debugElement = fixture.debugElement;
+    component = fixture.componentInstance;
+    store = TestBed.inject(Store);
+    actions$ = TestBed.inject(Actions);
+    component.workbasket = { ...selectedWorkbasketMock };
+    component.accessItemsRepresentation = workbasketAccessItemsMock;
+    store.reset({
+      ...store.snapshot(),
+      engineConfiguration: engineConfigurationMock,
+      workbasket: {
+        workbasketAccessItems: workbasketAccessItemsMock
+      }
+    });
+    fixture.detectChanges();
+  }));
+
+  afterEach(async(() => {
+    component.workbasket = { ...selectedWorkbasketMock };
+  }));
+
+  it('should create component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should initialize when accessItems exist', async((done) => {
+    component.action = ACTION.COPY;
+    let actionDispatched = false;
+    component.onSave = jest.fn().mockImplementation();
+    actions$.pipe(ofActionDispatched(GetWorkbasketAccessItems)).subscribe(() => (actionDispatched = true));
+    component.init();
+    expect(component.initialized).toBe(true);
+    expect(actionDispatched).toBe(true);
+    expect(component.onSave).toHaveBeenCalled();
+  }));
+
+  it("should discard initializing when accessItems don't exist", () => {
+    component.workbasket._links.accessItems = null;
+    component.init();
+    expect(component.initialized).toBe(false);
+  });
+
+  it('should add accessItems when add access item button is clicked', () => {
+    const addAccessItemButton = debugElement.nativeElement.querySelector('button.add-access-item');
+    const clearSpy = jest.spyOn(component, 'addAccessItem');
+    addAccessItemButton.click();
+    expect(addAccessItemButton.title).toMatch('Add new access');
+    expect(clearSpy).toHaveBeenCalled();
+  });
+
+  it('should undo changes when undo button is clicked', () => {
+    const undoButton = debugElement.nativeElement.querySelector('button.undo-button');
+    const clearSpy = jest.spyOn(component, 'clear');
+    undoButton.click();
+    expect(undoButton.title).toMatch('Undo Changes');
+    expect(clearSpy).toHaveBeenCalled();
+  });
+
+  it('should check all permissions when check all box is checked', () => {
+    const checkAllSpy = jest.spyOn(component, 'checkAll');
+    const checkAllButton = debugElement.nativeElement.querySelector('#checkbox-0-00');
+    checkAllButton.click();
+    expect(checkAllSpy).toHaveBeenCalled();
+    expect(checkAllButton).toBeTruthy();
```

Comment:
Is it better to call `expect(checkAllButton).toBeTruthy();` before `checkAllButton.click();`? In case this test fails, the click and the other test would have failed anyway.

Reply:
yes that makes sense

Reply:
you did not update this

Reply:
you did not update this